### PR TITLE
openjfx-modular-sdk, openjfx-sdk: add OpenJFX SDK

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

`openjfx-sdk` consists with JAR files and .so libraries for JavaFX (OpenJFX) modules.  This can be used with `adoptopenjdk-bin`, `zulu`, and `graalvm11-ce` like this:

```
javac --module-path path/to/javafx-sdk/lib/ \
  --add-modules javafx.graphics,javafx.controls,javafx.media,javafx.web JavaSample.java

java --module-path path/to/javafx-sdk/lib/ \
  --add-modules javafx.graphics,javafx.controls,javafx.media,javafx.web JavaSample
```

Existing `openjfx`, which is used to build `openjdk`, is renamed to `openjfx-modular-sdk` for consistency.

This patch also enables `openjdk` to run JavaFX apps with `MediaPlayer`.

See also #106716.

###### How to test

To test this PR, try https://gist.github.com/taku0/2c52b0b13876fb892a69c92c81e8f019.

This opens JavaFX windows with `adoptopenjdk-bin` (11 and 15), `zulu`, `graalvm11-ce`, and `openjdk` (11 and 15).
Clicking `Hello, JavaFX!` should play a movie.
Closing windows labeled `Hello, Swing!` terminates the application.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/openjdk/openjfx/11.nix [hunk 1]
```diff
@@ -67,47 +67,73 @@ let
...
+  sdk = common {
+    pname = "openjfx-sdk";
+
+    installPhase = ''
+      cp -r build/sdk $out
+    '';
+
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
+      license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/11.nix:133]: gpl2Only or gpl2Plus?

Comment (taku0) [reply pkgs/development/compilers/openjdk/openjfx/11.nix]: GPL v2 only, with Classpath exception

https://gluonhq.com/products/javafx/
https://github.com/openjdk/jfx
https://github.com/openjdk/jfx/blob/master/modules/javafx.base/src/main/java/javafx/beans/DefaultProperty.java

Updated the license info.

**Comment by taku0** [comment]
Fixed conflicts.

**Comment by taku0** [comment]
Updated to OpenJFX 16. The URL of OpenJFX 11 was broken and fixed.

File: pkgs/development/compilers/openjdk/openjfx/11.nix [hunk 2]
```diff
@@ -67,47 +67,73 @@ let
...
+    installPhase = ''
+      cp -r build/modular-sdk $out
+    '';
 
-  disallowedReferences = [ openjdk11-bootstrap ];
+    passthru.deps = deps;
 
-  passthru.deps = deps;
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/11.nix:116]:
```suggestion
      homepage = "https://openjdk.java.net/projects/openjfx/";
```

File: pkgs/development/compilers/openjdk/openjfx/11.nix [hunk 3]
```diff
@@ -67,47 +67,73 @@ let
...
+    '';
 
-  meta = with lib; {
-    homepage = "http://openjdk.java.net/projects/openjfx/";
-    license = licenses.gpl2;
-    description = "The next-generation Java client toolkit";
-    maintainers = with maintainers; [ abbradar ];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/11.nix:132]:
```suggestion
      homepage = "https://openjdk.java.net/projects/openjfx/";
```

File: pkgs/development/compilers/openjdk/openjfx/16.nix [hunk 4]
```diff
@@ -69,47 +69,73 @@ let
...
+    installPhase = ''
+      cp -r build/modular-sdk $out
+    '';
 
-  disallowedReferences = [ openjdk11_headless ];
+    passthru.deps = deps;
 
-  passthru.deps = deps;
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/16.nix:118]:
```suggestion
      homepage = "https://openjdk.java.net/projects/openjfx/";
```

File: pkgs/development/compilers/openjdk/openjfx/16.nix [hunk 5]
```diff
@@ -69,47 +69,73 @@ let
...
+    '';
 
-  meta = with lib; {
-    homepage = "http://openjdk.java.net/projects/openjfx/";
-    license = licenses.gpl2;
-    description = "The next-generation Java client toolkit";
-    maintainers = with maintainers; [ abbradar ];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/16.nix:134]:
```suggestion
      homepage = "https://openjdk.java.net/projects/openjfx/";
```

Comment (asbachb) [reply pkgs/development/compilers/openjdk/openjfx/16.nix:134]: AFAIK https://openjfx.io/ is now used as development space.

**Comment by taku0** [comment]
@SuperSandro2000 thank you for your review. Fixed.

File: pkgs/development/compilers/openjdk/openjfx/16.nix (hunk 5 above)
Comment (taku0) [reply pkgs/development/compilers/openjdk/openjfx/16.nix]: I'm not sure which URL is appropriate for the `homepage`.

- https://github.com/openjdk/jfx still points to https://openjdk.java.net/projects/openjfx/ from “About”.
- [README.md](https://github.com/openjdk/jfx/blob/5b42b642cefef3129c8925f5496d161ffb0ab285/README.md) says https://openjfx.io/ is the JavaFX community site.
- https://openjdk.java.net/projects/openjfx/ says “OpenJFX Home Page” is https://wiki.openjdk.java.net/display/OpenJFX/Main.
- https://wiki.openjdk.java.net/display/OpenJFX/Main says https://openjfx.io/ is the “OpenJFX Community Site”.

File: pkgs/development/compilers/openjdk/openjfx/11.nix [hunk 6]
```diff
@@ -15,8 +15,8 @@ let
     version = "${major}${update}-${build}";
 
     src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/11.nix:17]: Please use fetchFromGitHub.

Comment (taku0) [reply pkgs/development/compilers/openjdk/openjfx/11.nix]: Fixed.

**Comment by taku0** [comment]
Waiting for #125079.
`ffmpeg_3` is now deprecated. Newer OpenJFX is required to use `ffmpeg`. Newer OpenJFX requires newer Gradle, which is broken on i686. We cannot compute hash of `deps`.

**Comment by philiptaron** [comment]
I believe this PR is obviated on nixpkgs today; I see

* [`pkgs/development/compilers/openjdk/openjfx/11.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/11.nix)
* [`pkgs/development/compilers/openjdk/openjfx/15.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/15.nix)
* [`pkgs/development/compilers/openjdk/openjfx/17.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/17.nix)
* [`pkgs/development/compilers/openjdk/openjfx/19.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/19.nix)
* [`pkgs/development/compilers/openjdk/openjfx/20.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/20.nix)
* [`pkgs/development/compilers/openjdk/openjfx/21.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/21.nix)
* [`pkgs/development/compilers/openjdk/openjfx/22.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/openjfx/22.nix)

**Pull request closed without merging**